src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -23,28 +26,86 @@
 @WebServlet("/auth")
 public class AuthServlet extends HttpServlet {
 
-  // Boolean that holds the current sign in status.
-  private boolean signedIn;
-
   @Override
-  public void init() {
-    signedIn = false;
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    // Set up user auth objects.
+    UserService userService = UserServiceFactory.getUserService();
+    UserAuth userAuth;
+
+    // Set redirect URL after login / logout as index page.
+    final String redirectUrl = "/";
+
+    // Create UserAuth object with relevant login / logout information.
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String logoutUrl = userService.createLogoutURL(redirectUrl);
+      String id = userService.getCurrentUser().getUserId();
+
+      // Create UserAuth object to represent logged-in user.
+      userAuth = new UserAuth(logoutUrl, userEmail);
+    } else {
+      String loginUrl = userService.createLoginURL(redirectUrl);
+
+      // Create UserAuth object to represent logged-out user.
+      userAuth = new UserAuth(loginUrl);
+    }
+
+    // Create and send the JSON.
+    String json = convertToJson(userAuth);
+    response.getWriter().println(json);
   }
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Return the current sign in status.
-    response.setContentType("application/json;");
-    response.getWriter().println(signedIn);
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
   }
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Toggle signedIn value; temporary measure to show auth.
-    signedIn = !signedIn;
-    
-    // Redirect back to homepage.
-    response.sendRedirect("/");
+  /**
+  * Converts a UserAuth object into a JSON string using the Gson library.
+  */
+  private String convertToJson(UserAuth userAuth) {
+    Gson gson = new Gson();
+    String json = gson.toJson(userAuth);
+    return json;
   }
 
+  /**
+   * Inner class that holds relevant login/logout and user information.
+   */
+  class UserAuth {
+    // Fields that hold relevant login data.
+    private boolean loggedIn;
+    private String loginUrl;
+    private String logoutUrl;
+    private String email;
```

Comment:
I made a similar class in my portfolio and bootsie told me that:
- the loggedIn boolean is not necessary since we can always check whether the email is null or not
- correct me if im wrong. We are only ever going to need to have one link: either login or logout so we don't need 2 url links.
I'd me happy to show you my class if you wanted to take a look.

Reply:
Yeah we should talk tomorrow -- I have reasons for how I structured the class this way, though I'm open to changing it.

Reply:
Talked with Bootsie, and I took this advice!